great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -99,7 +99,7 @@ def __init__(self, *args, **kwargs):
 
         try:
             self._azure = BlobServiceClient(**azure_options)
-        except (TypeError, AttributeError):
+        except (NameError, TypeError, AttributeError):
             self._azure = None
```

Comment:
Thank you for fixing this! A few questions:
1. Should we also apply `NameError` to the previous S3 check?
2. Could the issue be with the way we've imported (Azure uses `except` rather than `except ImportError`)?

Reply:
I'm not sure why I didn't get the same NameError on the S3 check.

It's generally better to be specific on the type of error rather than just using a bare `except`

Reply:
Also, it might be worth reviewing whether your CI environment is including any other libraries that aren't being installed by Great Expectations' dependency requirements.

Reply:
Thanks for taking a look Dave! I think we've pinpointed the change in this PR: https://github.com/great-expectations/great_expectations/pull/3292

And the point about our CI environment is one that we will definitely take to heart. Some tests around clean installs of GE are on the horizon to make sure these kind of breaking changes are caught before release.

Thanks for your time and patience!